### ISSUE TITLE
optuna.create_study() throws MYSQL_OPT_RECONNECT warning

### ISSUE THREAD

**Issue by FlorinAndrei** [optuna/optuna#4955]
### Expected behavior

```python
OPTUNA_URL = 'mysql://optuna:XXXXXXXXX@127.0.0.1/optuna'
optuna.create_study(direction='minimize', study_name='foobar', storage=OPTUNA_URL, load_if_exists=True)
```

I get:

```
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
[I 2023-09-28 16:34:11,755] A new study created in RDB with name: foobar
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
```

### Environment

- Optuna version: 3.3.0
- Python version: 3.11.5
- OS: Linux-6.2.0-26-generic-x86_64-with-glibc2.35
- (Optional) Other libraries and their versions:

Ubuntu 22.04.3
MySQL 8.0.34

### Error messages, stack traces, or logs

```shell
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
[I 2023-09-28 16:34:11,755] A new study created in RDB with name: lgbmr
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
```

### Steps to reproduce

1. Replace XXXXXXXXX with the password
2. Run this code:
```python
OPTUNA_URL = 'mysql://optuna:XXXXXXXXX@127.0.0.1/optuna'
optuna.create_study(direction='minimize', study_name='foobar', storage=OPTUNA_URL, load_if_exists=True)
```

**Comment by HideakiImamura** [comment]
@FlorinAndrei Does this problem still occur with the latest Optuna v3.4?

**Comment by beantowel** [comment]
how can I turn this warning off?

**Comment by HideakiImamura** [comment]
I've done a bit of research and it seems that this warning is specific to MySQL's API and has nothing to do with Optuna. If you're considering a fix, [this](https://stackoverflow.com/questions/76875569/warning-mysql-opt-reconnect-is-deprecated-and-will-be-removed-in-a-future-versi) might be a helpful reference.

**Comment by FlorinAndrei** [comment]
@HideakiImamura I am using nothing but the Optuna CLI in a terminal here, version 3.4.0:

```
$ optuna studies --storage $OPTUNA_URL
```

Output:

```
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
+-------+---------------+----------+---------------------+
| name  | direction     | n_trials | datetime_start      |
+-------+---------------+----------+---------------------+
| xgbr  | ('MINIMIZE',) |        0 |                     |
| cbr   | ('MINIMIZE',) |        0 |                     |
| lgbmr | ('MINIMIZE',) |      163 | 2023-09-30 17:20:46 |
+-------+---------------+----------+---------------------+
```

Saying "it's not optuna, it's something else" doesn't seem right.

**Comment by nzw0301** [comment]
I agree with @HideakiImamura. This should be dealt with by SQL driver library, like https://github.com/sqlalchemy/sqlalchemy/discussions/10489.

**Comment by FlorinAndrei** [comment]
The bug appears to be fixed in the current version SQLAlchemy 2.0.23

https://github.com/sqlalchemy/sqlalchemy/issues/10492
